hw/ip/otbn/dv/otbnsim/sim/wsr.py

```diff
@@ -0,0 +1,128 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import List, Optional
+
+from riscvmodel.types import Trace  # type: ignore
+
+
+class TraceWSR(Trace):  # type: ignore
+    def __init__(self, wsr_name: str, new_value: int):
+        self.wsr_name = wsr_name
+        self.new_value = new_value
+
+    def __str__(self) -> str:
+        return '{} = {:#x}'.format(self.wsr_name, self.new_value)
+
+
+class WSR:
+    '''Models a Wide Status Register'''
+    def __init__(self, name: str):
+        self.name = name
+
+    def read_unsigned(self) -> int:
+        '''Get the stored value as a 256-bit unsigned value'''
+        raise NotImplementedError()
+
+    def write_unsigned(self, value: int) -> None:
+        '''Set the stored value as a 256-bit unsigned value'''
+        raise NotImplementedError()
+
+    def read_signed(self) -> int:
```

Comment:
Do we have any WSRs that can be negative? Certainly MOD and ACC are always positive (though I don't think this is explicitly documented anywhere). Where we will use read_signed instead of read_unsigned?

At the very least I'd argue that `read_signed` should return the same thing as `read_unsigned` (i.e. never return a negative number) for MOD and ACC and probably RND too.

Reply:
Well, I don't believe the behaviour should depend on what WSR we're reading. `read_unsigned` is supposed to be "read the 256 bits, interpreting it as a number in the range [0, 2^256-1]" and similarly for `read_signed`.